nx/lib/nx/defn/expr.ex

```diff
@@ -213,7 +213,14 @@ defmodule Nx.Defn.Expr do
   """
   def while(initial, context, arg, condition, body) do
     [flatten_initial, flatten_arg, flatten_body] = clauses = flatten_clauses([initial, arg, body])
-    args = [flatten_initial, flatten_arg, condition, flatten_body]
+
+    args = [
+      Nx.devectorize(flatten_initial, keep_names: false),
+      flatten_arg,
+      Nx.devectorize(condition, keep_names: false),
+      Nx.devectorize(flatten_body, keep_names: false)
+    ]
+
     flatten_to_composite(initial, context, clauses, &expr(&1, context, :while, args))
```

Comment:
We need to make sure the results of while are still vectorized if `flatten_initial` or `flatten_body` (they should match) are vectorized.

Reply:
the matching is done on another part of the code, and `flatten_to_composite` ensures the result is revectorized according to the first argument